website/docs/provisioners/habitat.html.markdown

```diff
@@ -47,20 +47,25 @@ There are 2 configuration levels, `supervisor` and `service`.  Configuration pla
 
 ### Supervisor Arguments
 * `version (string)` - (Optional) The Habitat version to install on the remote machine.  If not specified, the latest available version is used.
-* `use_sudo (bool)` - (Optional) Use `sudo` when executing remote commands.  Required when the user specified in the `connection` block is not `root`.  (Defaults to `true`)
+* `license (string)` - (Optional) Sets the acceptance of Chef licensing: https://www.chef.io/end-user-license-agreement/
+* `auto_update (bool)` - (Optional) If set to `true`, the supervisor will auto-update itself as soon as new releases are available on the specified `channel`.
+* `http_disable (bool)` - (Optional) If set to `true`, disables the supervisor HTTP listener entirely.
+* `peers (array)` - (Optional) A list of IP or FQDN's of other supervisor instance(s) to peer with. (Defaults to none)
 * `service_type (string)` - (Optional) Method used to run the Habitat supervisor.  Valid options are `unmanaged` and `systemd`.  (Defaults to `systemd`)
 * `service_name (string)` - (Optional) The name of the Habitat supervisor service, if using an init system such as `systemd`. (Defaults to `hab-supervisor`)
-* `peer (string)` - (Optional) IP or FQDN of a supervisor instance to peer with. (Defaults to none)
+* `use_sudo (bool)` - (Optional) Use `sudo` when executing remote commands.  Required when the user specified in the `connection` block is not `root`.  (Defaults to `true`)
 * `permanent_peer (bool)` - (Optional) Marks this supervisor as a permanent peer.  (Defaults to false)
+* `listen_ctl (string)` - (Optional) The listen address for the countrol gateway system (Defaults to 127.0.0.1:9632)
 * `listen_gossip (string)` - (Optional) The listen address for the gossip system (Defaults to 0.0.0.0:9638)
 * `listen_http (string)` - (Optional) The listen address for the HTTP gateway (Defaults to 0.0.0.0:9631)
 * `ring_key (string)` - (Optional) The name of the ring key for encrypting gossip ring communication (Defaults to no encryption)
 * `ring_key_content (string)` - (Optional) The key content.  Only needed if using ring encryption and want the provisioner to take care of uploading and importing it.  Easiest to source from a file (eg `ring_key_content = "${file("conf/foo-123456789.sym.key")}"`) (Defaults to none)
+* `ctl_secret (string)` - (Optional) Specify a secret to use (from `hab sup secret generate`) for control gateway communication between hab client(s) and the supervisor.  (Defaults to none)
 * `url (string)` - (Optional) The URL of a Builder service to download packages and receive updates from.  (Defaults to https://bldr.habitat.sh)
 * `channel (string)` - (Optional) The release channel in the Builder service to use. (Defaults to `stable`)
 * `events (string)` - (Optional) Name of the service group running a Habitat EventSrv to forward Supervisor and service event data to. (Defaults to none)
-* `override_name (string)` - (Optional) The name of the Supervisor (Defaults to `default`)
```

Comment:
Why remove this option?

Reply:
My understanding is that the `override_name` option is only for cases where you have multiple hab supervisors running on a single node and you need to point them at more than one state directory (e.g., one at `/hab/sup/default`, another at `/hab/sup/custom`, etc.).  If someone did specify `override_name` in the terraform options like `custom`, none of the resulting configuration metadata for that supervisor instance was written to the correct dir (paths were hard-coded with `/hab/sup/default/CTL_SECRET`, `/hab/sup/default/sup.log`, etc.).  As a result, it wouldn't actually work, and I'm guessing nobody used that option. It would probably be confusing to anyone trying to use the provisioner for the first time, so I opted to remove it (but, I can add it back if you think that's best).

Reply:
That's very helpful context! Thank you for that. It's generally a good question to ask when something is removed, and your reasoning here seems solid to me as far as allowing people to open an issue if this is something they had working somehow.